polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -685,7 +685,7 @@ impl<Config: config::Config> XcmExecutor<Config> {
 		remote_xcm: &mut Vec<Instruction<()>>,
 	) -> Result<Assets, XcmError> {
 		// Must ensure that we recognise the assets as being managed by the destination.
-		#[cfg(not(feature = "runtime-benchmarks"))]
+		#[cfg(not(any(test, feature = "runtime-benchmarks")))]
```

Comment:
I think it's fine to skip these checks for benchmarks, but if we skip them for tests as well, we won't be able to guarantee, test, or verify that it's working (the runtime, xcm-executor) as expected.

Reply:
they are skipped for local crate tests, this stuff is covered/tested in emulator where it is enabled

we could allow it in local tests too but then would have to adapt the mock and the tests - that would be the right thing to do, but takes some time and doesn't bring much value - maybe open a low-prio issue for follow-up?

Reply:
ok, ok, let's do a follow-up: https://github.com/paritytech/polkadot-sdk/issues/6271